plugins/filter_aws/aws.c

```diff
@@ -370,6 +385,134 @@ static int get_vpc_metadata(struct flb_filter_aws *ctx)
     return ret;
 }
 
+/* Get EC2 instance tag keys from /latest/meta-data/tags/instance.
+ * Initializes ctx->tags_count, ctx->tag_keys and ctx->tag_keys_len.
+ */
+static int get_ec2_tag_keys(struct flb_filter_aws *ctx)
+{
+    int ret;
+    flb_sds_t tags_list = NULL;
+    size_t len = 0;
+    size_t tag_index = 0;
+    size_t tag_start = 0;
+    size_t tag_end = 0;
+    size_t characters_to_copy;
+    size_t i;
+
+    /* get a list of tag keys from the meta data server */
+    ret = get_metadata(ctx, FLB_FILTER_AWS_IMDS_INSTANCE_TAG, &tags_list, &len);
+    if (ret < 0) {
+        if (ret == -2) {
+            flb_warn("[filter_aws] tags not available in the EC2 instance metadata");
+            ctx->tags_count = 0;
+            return 0;
+        }
+        flb_sds_destroy(tags_list);
+        return -1;
+    }
+    /* count number of tag keys and allocate memory */
+    /* it assumes there always is at least one tag */
+    /* starts with 1, because '\n' is a split character */
+    ctx->tags_count = 1;
+    for (i = 0; i < len; i++) {
+        if (tags_list[i] == '\n') {
+            ctx->tags_count++;
+        }
+    }
+    ctx->tag_keys = flb_calloc(ctx->tags_count, sizeof(flb_sds_t*));
+    ctx->tag_keys_len = flb_calloc(ctx->tags_count, sizeof(size_t*));
+
+    /* go over the response and initialize tag_keys values */
+    /* code below finds two indices which define tag key and copies them to ctx */
+    while (tag_end <= len) {
+        if (tags_list[tag_end] == '\n') {
+            tags_list[tag_end] = '\0';
+        }
+        if ((tags_list[tag_end] == '\0' || tag_end == len) && (tag_start < tag_end)) {
+            /* length of the key + 1 place for null character */
+            characters_to_copy = tag_end - tag_start;
+
+            ctx->tag_keys_len[tag_index] = characters_to_copy;
+            ctx->tag_keys[tag_index] = flb_sds_create_size(characters_to_copy + 1);
+            if (ctx->tag_keys[tag_index] == NULL) {
+                flb_sds_destroy(tags_list);
+                return -2;
+            }
+
+            flb_sds_t tag_key = tags_list + tag_start;
```

Comment:
Always define variable at top of functions: https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md

Reply:
Thanks! I missed this one. https://github.com/fluent/fluent-bit/pull/6065/commits/302a77f53921b30ef1628a78efec38b1099618f0

Reply:
Conversation resolved due to lack of new activity. (Feel free to bring it back.)